docs/sql-performance-tuning.md

```diff
@@ -428,3 +428,101 @@ You can control the details of how AQE works by providing your own cost evaluato
       <td>3.2.0</td>
     </tr>
   </table>
+
+## Storage Partition Join
```

Comment:
Can we have at least one working example? e.g., how do we use it with Parquet DSv2?

Reply:
Thanks for review.  The mechanism works today if DSV2 partitioning implements KeyGroupedPartitioning. 
 Unfortunately the initial SPJ implementation did not support it for ParquetDSV2, so I think there's no data source in Spark code that we can have an example at the moment (the tests use the InMemoryTable which implements it).  I can potentially investigate implementing for ParquetDSV2 later.

Reply:
Could we have an example with thridparty library that is open source?

Reply:
@HyukjinKwon sure, I put iceberg example, let me know what you think